airbyte-decorator.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime, timedelta
+from typing import Any, Callable, Mapping, Optional
+
+from dagster import (
+    AssetExecutionContext,
+    Definitions,
+    DefinitionsLoadContext,
+    DefinitionSource,
+    DefinitionsReloadRequest,
+    JsonMetadataValue,
+    SensorResult,
+    build_last_update_freshness_checks,
+    defs_loader,
+    sensor,
+)
+
+####################################################################################################
+# This is code that would live inside the dagster-airbyte library
+####################################################################################################
+
+
+class AirbyteWorkspace:
+    def __init__(self, name: str):
+        self._name = name
+
+    def build_defs(
+        self, context: DefinitionsLoadContext, execution_fn: Optional[Callable] = None
+    ) -> Definitions:
+        existing_source: DefinitionSource = context.get_loaded_definition_source(self._name)
```

Comment:
Where is get_loaded_definition_source defined?

Reply:
Kind of the crux of the whole thing

Reply:
I'm imagining that we would get it in the same way we get the `repository_load_data` used for `CacheableAssetsDefinition`. I.e. the code server places puts it in the execution plan snapshot when it creates a run, and the step / run worker picks it up from there.

I'm not certain that's the best way to store/transfer it in the long run, but I'm imagining that it would be a black box from the perspective of the user, with the guarantee being that the value provided to a step worker within run X is the same value that was loaded inside the code server that launched the run.